android/game/src/main/java/com/farsitel/bazaar/game/GameHubBridge.java

```diff
@@ -0,0 +1,422 @@
+        isLogin(context, false, loginResult -> {
+            if (loginResult.status != Status.SUCCESS) {
+                getTournamentRankingCallback(loginResult, callback);
+                return;
+            }
+
+            if (isBroadcastMode) {
+                gameHubBroadcast.getCurrentLeaderboard(rankResult -> {
+                    getTournamentRankingCallback(rankResult, callback);
+                });
+                return;
+            }
+            executorService.submit(() -> {
+                final Result result = new Result();
+                try {
+                    Bundle bundle = gameHubService.getCurrentLeaderboard(context.getPackageName());
+                    result.setBundle(bundle);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    result.status = Status.FAILURE;
+                    result.message = e.getMessage();
+                    result.stackTrace = Arrays.toString(e.getStackTrace());
+                }
+                MainThread.run(() -> getTournamentRankingCallback(result, callback));
+            });
+        });
+    }
+
+    void getTournamentRankingCallback(Result result, IRankingCallback callback) {
+        logger.logDebug("Ranking: " + result.toString());
+
+        if (result.status != Status.SUCCESS) {
+            callback.onFinish(result.status.getLevelCode(), result.message, result.stackTrace, null);
+            return;
+        }
+
+        List<RankItem> rankItems = new ArrayList<>();
+        JSONArray participants = null;
+        String jsonString = result.extras.getString(ParamNames.LEADERBOARD_DATA);
+        try {
+            JSONObject jsonObject = new JSONObject(jsonString);
+            participants = jsonObject.optJSONArray(ParamNames.PARTICIPANTS);
+            int participantsCount = participants != null ? participants.length() : 0;
+            for (int i = 0; i < participantsCount; i++) {
+                JSONObject obj = participants.getJSONObject(i);
+                rankItems.add(new RankItem(obj.getString(ParamNames.NICKNAME),
+                        obj.getString(ParamNames.SCORE),
+                        obj.getString(ParamNames.AWARD),
+                        obj.getBoolean(ParamNames.HAS_FOLLOWING_ELLIPSIS),
+                        obj.getBoolean(ParamNames.IS_CURRENT_USER),
+                        obj.getBoolean(ParamNames.IS_WINNER)));
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
+            callback.onFinish(Status.FAILURE.getLevelCode(), "Error on Ranking data parsing!", "", null);
+            return;
+        }
+
+        callback.onFinish(result.status.getLevelCode(), "getTournamentRanking", jsonString, rankItems);
+    }
+
+    @Override
+    public boolean disposed() {
+        if (gameHubBroadcast != null) {
+            gameHubBroadcast.dispose();
+        }
+
+        if (executorService != null) {
+            executorService.shutdown();
```

Comment:
After shutdown set null it and move initializing executorService from body of class into constructor to reinitialize every time GameHubBridge is creating